### ISSUE TITLE
REST API: Support sorting by specified field

### ISSUE THREAD

**Issue by Bosch-Eli-Black** [keycloak/keycloak#14189]
### Description

The REST API (https://github.com/keycloak/keycloak-community/blob/main/design/rest-api-guideline.md) supports pagination and filtering, but not sorting.

Being able to sort by a column would make pagination more useful, as the user could sort by a non-default column and request only one page of results.

For example, consider a table where a user can click on a column header to sort the table by that column:

| Username | Email | First name | Last name |
| ----------- | ------ | ------------ | ---------- |
| awesomeguy183 | joe@example.com | Joey | Brown |
| happy_girl | Joanna@example.com | Joanna | Green |

When someone clicks the "Last name" column header, the table should be sorted by the last name. The table is divided into several pages of results, because there are many users in the system.

Without being able to sort by the "Last name" field via the REST API, the frontend needs to retrieve the data for all users, and sorting performed via the frontend. This can be quite a bit of data if the Keycloak has a lot of users.

If we could sort via the backend, the frontend would only need to download a small subset of the results 🙂

**Comment by JoseOliveira01** [comment]
It's probably not the best approach but I'm using a microservice called user-management to address this issue. I'm connecting to a secondary data source (keycloak's schema) and sorting the users using my own parameters. 
However, if this feature came out, I would appreciate it!

**Comment by inimaz** [comment]
I agree, this feature would be very helpful. 

In the meantime, I am doing same as @JoseOliveira01. The problem with this microservice approach is that in order to perform sorting, one has to get the whole list of users in the memory of the microservice and then return the list. This is not scalable if one has lots of users. Plus we cannot use keycloak's pagination.

**Comment by riotsnotdiets** [comment]
Indeed, how do you use "first" and "max" for i.e. users when you dont have a sort? Imagine a list of users sorted by name and you want the second page .. you dont get the second page of the sorted users but the second page of the default sorting. You cant sort if afterwards, because the second page is already "wrong".

Finally, you have to get each and every user and paginate and sort yourself?

**Comment by oliveirapt** [comment]
I have sorting already. I just don't like the idea of having 2 services (keycloak and my user-management service) connected to the exact same database/schema to address this issue. 

I was using the user-management to get sorted chunks from a secondary postgres database where my keycloak is currently "connected". The service handles a query with the size and offset that i want. 

Something like the following:

```mermaid
graph TD
    UM[User-Management Service] -->|Reads & Writes| DB1[User-Management Database]
    UM -->|Fetches Users using pagination/sort| KCDB[Keycloak Database]
    KC[Keycloak] -->|Reads & Writes| KCDB[Keycloak Database]

**Comment by koplas** [comment]
Are there any proposals on how the API should look like? It can also be helpful also to be able to sort by custom user profile attributes.

**Comment by koplas** [comment]
I opened a pull request [#41280](https://github.com/keycloak/keycloak/pull/41280) to explore how the API should look like. Are there any guidelines that need to be considered?
